tests/nosetests/pyanaconda_tests/module_subscription_tasks_tests.py

```diff
@@ -241,3 +246,182 @@ def set_rhsm_config_tast_restore_default_value_test(self):
+        """Test the stop_rhsm_private_bus method."""
+        # mock the register server proxy
+        register_server_proxy = Mock()
+        # make sure locate is always the same at test time
+        environ_get.return_value = "en_US.UTF-8"
+        # call the method
+        stop_rhsm_private_bus(register_server_proxy)
+        # check the proxy was called appropriately
+        register_server_proxy.Stop.assert_called_once_with("en_US.UTF-8")
+
+
+class RegistrationTasksTestCase(unittest.TestCase):
+    """Test the registration tasks."""
+
+    @patch("os.environ.get")
+    @patch("pyanaconda.modules.subscription.runtime.stop_rhsm_private_bus")
+    @patch("pyanaconda.modules.subscription.runtime.start_rhsm_private_bus")
+    def username_password_success_test(self, start_private_bus, stop_private_bus, environ_get):
+        """Test the RegisterWithUsernamePasswordTask - success."""
+        # register server proxy
+        register_server_proxy = Mock()
+        # private register proxy
+        private_register_proxy = start_private_bus.return_value
+        private_register_proxy.Register.return_value = True, ""
+        # make sure locate is always the same at test time
+        environ_get.return_value = "en_US.UTF-8"
+        # instantiate the task and run it
+        task = RegisterWithUsernamePasswordTask(rhsm_register_server_proxy=register_server_proxy,
+                                                username="foo_user",
+                                                password="bar_password")
+        success, error_message = task.run()
+        # check the return values
+        self.assertTrue(success)
+        self.assertEqual(error_message, "")
+        # check start_private_bus was called
+        start_private_bus.assert_called_once_with(register_server_proxy)
+        # check private register proxy Register method was called
+        private_register_proxy.Register.assert_called_with('',
+                                                           'foo_user',
+                                                           'bar_password',
+                                                           {},
+                                                           {},
+                                                           'en_US.UTF-8')
+        # and finally, the private bus should be stopped
+        stop_private_bus.assert_called_once_with(register_server_proxy)
+
+    @patch("os.environ.get")
+    @patch("pyanaconda.modules.subscription.runtime.stop_rhsm_private_bus")
+    @patch("pyanaconda.modules.subscription.runtime.start_rhsm_private_bus")
+    def username_password_failure_test(self, start_private_bus, stop_private_bus, environ_get):
+        """Test the RegisterWithUsernamePasswordTask - failure."""
+        # register server proxy
+        register_server_proxy = Mock()
+        # private register proxy
+        private_register_proxy = start_private_bus.return_value
```

Comment:
I suspect this in fact reads from the mock API (`return_value`) a thing that is unset, and so evaluates to just another `Mock()`. That would mean the line is misleading, as it implies some relationship where there is none. But I am not 100% sure.

The pattern is repeated everywhere.

Reply:
Yeah, I have been looking a bit more into how the `Mock` objects work and it really looks like `return_value` is always another Mock instance that gets cached and stays the same after fist access. So getting the mock instance from `return_value` should be equivalent to setting your own.

Anyway, cleaned up that part when changing the unit tests to work with `RHSMPrivateBus`.